### PR TITLE
feat-221 Add ability to edit tasks closes 221

### PR DESCRIPTION
**Pull request by PreetPatel**
- [x] The pull request is complete according to the following criteria:
  - [ ] Acceptance criteria have been met
  - [ ] The documentation is kept up-to-date
  - [ ] Comprehensive tests (if applicable) have been generated and all pass.
  - [ ] The pull request describes the changes that have been made, and enough information is present in the description for any developer to understand what has changed
  - [ ] Commits have been squashed (or will be on merge).
  - [ ] The branch name is descriptive and follows the pull request title format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
  - [ ] The pull request title is of the following format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
  - [ ] The description uses [github syntax](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) to link to the issue. E,g Resolves se701g2/Doto#{Number}
  - [ ] At least two reviewers assigned. One of which must be the assigner of the issue.
  - [ ] If there are merge conflicts, run git rebase as opposed to git merge with master.

### CODE REVIEW

**Comment by Minus20Five** [comment]
Hi @PreetPatel, 

I tried to test this locally but when I click on the edit button the page freezes and I get the following error in console.

**OS**: macOS
**Browser**: Chrome

image

**Comment by PreetPatel** [comment]
> Hi @PreetPatel,
> 
> I tried to test this locally but when I click on the edit button the page freezes and I get the following error in console.
> 
> **OS**: macOS
> **Browser**: Chrome
> 
> image

oh dear! thanks @Minus20Five ill have a look

File: doto-frontend/src/components/pages/Calendar/CalendarComponent.js [hunk 1]
```diff
@@ -52,47 +92,85 @@ export function Content ({ children, appointmentData, style, onTaskStatusUpdated
...
 
-    const deleteTask = (taskId) => {
+    const deleteTask = taskId => {
         onTaskDeleted(taskId);
         document.getElementById("grid").click();
     };
 
+    const handleOpenUpdateModal = () => {
+        setOpenUpdateModal(true);
+    };
```
Comment (salma-s) [inline doto-frontend/src/components/pages/Calendar/CalendarComponent.js:102]: There is a bug where the form for editing tasks doesn't get updated. To reproduce:
1. Click on a task. The tooltip will appear.
2. Click the edit button, edit the title and save your changes. You'll notice that the appointment tooltip is still open with the old title.
3. Click the edit button again, and you'll notice the edit form hasn't been updated.

To fix this, I think the tooltip has to be closed after clicking the `UPDATE` button. The Appointment.Tooltip component doesn't come with an out-of-the-box method of closing it. 

I suggest you copy how task deletion forcibly closes the tooltip in the `Content` function above.

**Comment by PreetPatel** [comment]
> Are you able to hide the scroll bar in the update task window when it's not needed?
> Functionality all seems to work fine 🙂

This is an additional feature... fixing your change requires changes to a modal that this feature did not create
So im happy to say - wont fix? 👀

**Comment by jordansimsmith** [comment]
@all-contributors please add @PreetPatel  for code, review, design, doc

**Comment by PreetPatel** [comment]
Holy shit that’s cool!

**Pull request merged**